### PR TITLE
Add CodeQL pipeline and make repo compliant

### PR DESCRIPTION
**Pull request by garath**
This adds a CodeQL pipeline scheduled for a weekly analysis. This also makes changes to resolve existing errors.

New pipeline is [arcade-services-codeql](https://dev.azure.com/dnceng/internal/_build?definitionId=1084). Latest green build (from the head of this PR) is build [1627220](https://dev.azure.com/dnceng/internal/_build/results?buildId=1627220&view=results).

Resolves dotnet/core-eng#15329.

### CODE REVIEW

File: azure-pipeline-codeql.yml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        
+      - pwsh: |
+          [xml]$manifest = Get-Content src\Maestro\MaestroApplication\ApplicationPackageRoot\ApplicationManifest.xml
+          $manifest.SelectSingleNode("/*[local-name()='ApplicationManifest']/*[local-name()='Policies']").RemoveAll()
+          $manifest.SelectSingleNode("/*[local-name()='ApplicationManifest']/*[local-name()='Principals']").RemoveAll()
+          $manifest.Save("src\Maestro\MaestroApplication\ApplicationPackageRoot\ApplicationManifest.xml")
+          git diff
+        displayName: Remove Service Fabric RunAsPolicy from MaestroApplication
+
+      - pwsh: |
```
Comment (MattGal) [inline azure-pipeline-codeql.yml:78]: Maybe it's obvious and I'm just unfamiliar with CodeQL but why are you modifying XML files as part of CodeQL build?

File: azure-pipeline-codeql.yml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+          . $(Build.SourcesDirectory)\eng\codeql.ps1
+          Initialize-Gdn -GuardianCliLocation $(GuardianCliLocation) `
+            -WorkingDirectory $(Build.SourcesDirectory) `
+            -LoggerLevel 'Standard'
+        displayName: 'CodeQL: Initialize'
+
+      - pwsh: |
+          . $(Build.SourcesDirectory)\eng\codeql.ps1
+          New-GdnSemmelConfig -GuardianCliLocation $(GuardianCliLocation) `
```
Comment (MattGal) [inline azure-pipeline-codeql.yml:95]: Some places, it's Semmle, here it's Semmel... looks like a function naming problem

File: eng/codeql.ps1 [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    )
+
+    if (!(Test-Path $GuardianCliLocation)) {
+        throw "GuardianCliLocation not found"
+    }
+
+    & $GuardianCliLocation init --working-directory $WorkingDirectory --logger-level $LoggerLevel
+}
+
+function New-GdnSemmelConfig {
```
Comment (MattGal) [inline eng/codeql.ps1:28]: Nit:  Should this be "New-GdnSemmleConfig"?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -50,7 +50,7 @@ stages:
         - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
           # DotNet-Blob-Feed provides: dotnetfeed-storage-access-key-1
           # DotNet-Symbol-Server-Pats provides: microsoft-symbol-server-pat, symweb-symbol-server-pat
-          # Publish-Build-Assets provides: MaestroAccessToken, BotAccount-dotnet-maestro-bot-PAT
+          # Publish-Build-Ass['=-ets provides: MaestroAccessToken, BotAccount-dotnet-maestro-bot-PAT
```
Comment (MattGal) [inline azure-pipelines.yml:53]: Typo?  If not typo, maybe add comment why this is here.

File: azure-pipeline-codeql.yml [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+            -LoggerLevel 'Standard' `
+            -Language 'csharp' `
+            -WorkingDirectory $(Build.SourcesDirectory) `
+            -SourceCodeDirectory $(Build.SourcesDirectory)\src `
+            -BuildCommand "$(Build.SourcesDirectory)\build.cmd -configuration Release -prepareMachine /p:Test=false /P:Sign=false # rmdir /s /q $(Build.SourcesDirectory)\src\Maestro\maestro-angular\dist" `
```
Comment (MattGal) [inline azure-pipeline-codeql.yml:100]: What's the commented-out rmdir here?

File: eng/codeql.ps1 [hunk 6]
```diff
@@ -0,0 +1,195 @@
+function Install-Gdn {
+    param(
+        [string]$Path
+    )
+
+    Start-Process nuget -Verbose -ArgumentList "install", "Microsoft.Guardian.Cli", "-Source https://securitytools.pkgs.visualstudio.com/_packaging/Guardian/nuget/v3/index.json", "-OutputDirectory $Path", "-NonInteractive", "-NoCache" -NoNewWindow -Wait
```
Comment (MattGal) [inline eng/codeql.ps1:6]: Remind me why this is Start-Process and not just a straight up nuget command?

File: azure-pipeline-codeql.yml (hunk 2 above)
Comment (garath) [reply azure-pipeline-codeql.yml:95]: Yes, my spelling is bad apparently. Will fix.

File: eng/codeql.ps1 (hunk 3 above)
Comment (garath) [reply eng/codeql.ps1:28]: Yep. Spelling 🔤.

File: azure-pipeline-codeql.yml (hunk 5 above)
Comment (garath) [reply azure-pipeline-codeql.yml:100]: That's actually a line delimiter in this context. CodeQL will execute both lines in a batch file separately. 

The usual build process has a little path that generates javascript for the Maestro Angular app. CodeQL thinks it is regular code and blames me for its problems. CodeQL doesn't want generated code when it analyzes the Angular app, so I just delete it here.

File: azure-pipelines.yml (hunk 4 above)
Comment (garath) [reply azure-pipelines.yml:53]: Definitely a typo.

File: azure-pipeline-codeql.yml (hunk 1 above)
Comment (garath) [reply azure-pipeline-codeql.yml:78]: This exists in the regular azure-pipelines.yaml and is involved in building the source. I just duplicated it here to ensure CodeQL is close to the "real" build.

**Review by MattGal — APPROVED** [review]

**Pull request merged**